src/base/orgs/View.svelte

```diff
@@ -191,67 +169,66 @@
         <!-- Name -->
         <div class="label">Name</div>
         <div>
-          {#await org.lookupAddress(config)}
-            <div class="loading"><Loading small /></div>
-          {:then name}
-            {#if name}
-              <Link to={`/registrations/${label}`}>{name}</Link>
-            {:else}
-              <span class="subtle">Not set</span>
-            {/if}
-          {/await}
+          {#if profile.name()}
+            <Link to={`/registrations/${parseEnsLabel(profile.name() ?? "", config)}`}>{profile.name()}</Link>
```

Comment:
Not sure about this: in what cases exactly are we returning `""`?

Reply:
Yeah the issue here is that `profile.name()` returns `string | undefined` and `parseEnsLabel` demands a string.
We could eventually update `parseEnsLabel` and the functions that use said function to be able to handle a `undefined` value.
Or the other option would be to resolve the Profile in an onMount function and pass it to the components already resolved.

Reply:
Ok I pushed a commit with a posible solution, it's not the nicest because it starts replacing `null` with `undefined` and does it only on two individual functions.. And it also changes the behaviour of these two functions. b4dfa25963be687eff8d208543abd74da10b4212
Let me know what you think, and as another option we could do the resolving in an onMount function

Reply:
Yeah I don't like the solution in b4dfa25 because we lose type safety: the compiler will not error if we pass an undefined variable in the function..

Reply:
It's annoying because the `{#if profile.name()}` should provide the type assertion that the value is not `undefined`, but it seems like svelte-check is not smart enough here..

Reply:
It seems like switching to accessors for the Profile class gets around this issue:

```diff
diff --git a/src/profile.ts b/src/profile.ts
index 9c6049a..6cf8bd1 100644
--- a/src/profile.ts
+++ b/src/profile.ts
@@ -39,7 +39,7 @@ export class Profile {
   }

   // Using undefined as return type if nothing to be returned since it works better with <a href> links
-  name(): string | undefined {
+  get name(): string | undefined {
     if (this.profile?.ens?.name) return this.profile.ens.name;
     else if (this.profile?.idx?.name) return this.profile.idx.name;
     else return undefined;
@@ -169,8 +169,8 @@
         <!-- Name -->
         <div class="label">Name</div>
         <div>
-          {#if profile.name()}
-            <Link to={`/registrations/${parseEnsLabel(profile.name(), config)}`}>{profile.name()}</Link>
+          {#if profile.name}
+            <Link to={`/registrations/${parseEnsLabel(profile.name, config)}`}>{profile.name}</Link>
           {:else}
```

Reply:
> It seems like switching to accessors for the Profile class gets around this issue:
> 
> ```diff
> diff --git a/src/profile.ts b/src/profile.ts
> index 9c6049a..6cf8bd1 100644
> --- a/src/profile.ts
> +++ b/src/profile.ts
> @@ -39,7 +39,7 @@ export class Profile {
>    }
> 
>    // Using undefined as return type if nothing to be returned since it works better with <a href> links
> -  name(): string | undefined {
> +  get name(): string | undefined {
>      if (this.profile?.ens?.name) return this.profile.ens.name;
>      else if (this.profile?.idx?.name) return this.profile.idx.name;
>      else return undefined;
> @@ -169,8 +169,8 @@
>          <!-- Name -->
>          <div class="label">Name</div>
>          <div>
> -          {#if profile.name()}
> -            <Link to={`/registrations/${parseEnsLabel(profile.name(), config)}`}>{profile.name()}</Link>
> +          {#if profile.name}
> +            <Link to={`/registrations/${parseEnsLabel(profile.name, config)}`}>{profile.name}</Link>
>            {:else}
> ```

Ooh wow that's great! Ty will do the refactor right away